src/main/java/com/esri/samples/editing/edit_and_sync_features/EditAndSyncFeaturesSample.java

```diff
@@ -0,0 +1,470 @@
+                }
+              } catch (InterruptedException | ExecutionException e) {
+                displayMessage("Exception getting clicked feature", e.getMessage());
+              }
+            });
+          });
+
+        } else if (event.getButton() == MouseButton.SECONDARY) {
+
+          // on secondary mouse click, clear feature selection
+          for (Layer layer : mapView.getMap().getOperationalLayers()) {
+            FeatureLayer featureLayer = (FeatureLayer) layer;
+            featureLayer.clearSelection();
+          }
+        }
+      });
+
+      // add map view, controlsVBox and progressBar to stack pane
+      stackPane.getChildren().addAll(mapView, controlsVBox, progressIndicator);
+      StackPane.setAlignment(controlsVBox, Pos.TOP_LEFT);
+      StackPane.setMargin(controlsVBox, new Insets(10, 0, 0, 10));
+
+    } catch (Exception e) {
+      // on any error, display the stack trace
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Generates a local geodatabase and sets it to the map.
+   */
+  private void generateGeodatabase() {
+
+    // define geodatabase sync task
+    geodatabaseSyncTask = new GeodatabaseSyncTask("https://sampleserver6.arcgisonline.com/arcgis/rest/services/Sync/WildfireSync/FeatureServer");
+    geodatabaseSyncTask.loadAsync();
+    geodatabaseSyncTask.addDoneLoadingListener(() -> {
+
+      // create generate geodatabase parameters for the current extent
+      final ListenableFuture<GenerateGeodatabaseParameters> defaultParametersFuture = geodatabaseSyncTask
+              .createDefaultGenerateGeodatabaseParametersAsync(geodatabaseExtentGraphics.getGeometry());
+      defaultParametersFuture.addDoneListener(() -> {
+        try {
+          // set parameters and don't include attachments
+          GenerateGeodatabaseParameters defaultParameters = defaultParametersFuture.get();
+          defaultParameters.setReturnAttachments(false);
+
+          // create a temporary file for the geodatabase
+          File tempFile = File.createTempFile("gdb",".geodatabase");
+          tempFile.deleteOnExit();
+
+          // create and start the job
+          GenerateGeodatabaseJob geodatabaseJob = geodatabaseSyncTask.generateGeodatabase(defaultParameters, tempFile.getAbsolutePath());
+          geodatabaseJob.start();
+
+          // get geodatabase when done
+          geodatabaseJob.addJobDoneListener(() -> {
+            if (geodatabaseJob.getStatus() == Job.Status.SUCCEEDED) {
+              geodatabase = geodatabaseJob.getResult();
+              geodatabase.loadAsync();
+              geodatabase.addDoneLoadingListener(() -> {
+                if (geodatabase.getLoadStatus() == LoadStatus.LOADED) {
+
+                  // add the geodatabase FeatureTables to the map as a FeatureLayer
+                  geodatabase.getGeodatabaseFeatureTables().forEach(geodatabaseFeatureTable -> {
+                    mapView.getMap().getOperationalLayers().add(new FeatureLayer(geodatabaseFeatureTable));
```

Comment:
Can remove the curly braces here to make an expression lambda (as discussed for ClosestFacilityStatic sample). Can also reduce the code slightly by just calling
`map.getOperationalLayers().add(new FeatureLayer(geodatabaseFeatureTable));`

Reply:
Another good spot, thanks!